km/km_vcpu_run.c

```diff
@@ -509,6 +510,9 @@ static void km_vcpu_one_kvm_run(km_vcpu_t* vcpu)
    }
 
    if (rc != 0) {
+      if (errno == 0) {
```

Comment:
This does need a comment to the effect that we are compensating for some kind of bug in ioctl()'s ability to set errno properly.

Reply:
errno most probably has changed between line 499 - 513(depending on what options we run with). we need to save it between lines 499 and 500.

Reply:
Then this needs to be fixed differently.
Save errno after ioctl() returns and use that value to restore errno.

Plus ioctl() should not return -errno on failure but that is apparently what we see.

Reply:
`int
__ioctl (int fd, unsigned long int request, ...)
{
  va_list args;
  va_start (args, request);
  void *arg = va_arg (args, void *);
  va_end (args);

  int r;
  if (!__ioctl_arch (&r, fd, request, arg))
    {
      r = INTERNAL_SYSCALL_CALL (ioctl, fd, request, arg);
      if (__glibc_unlikely (INTERNAL_SYSCALL_ERROR_P (r)))
	{
	  __set_errno (-r);
	  return -1;
	}
    }
  return r;
}`